packages/cactus-api-client/src/main/typescript/api-client.ts

```diff
@@ -130,9 +132,19 @@ export class ApiClient extends BaseAPI {
 
     const randomNode = nodes[randomIdx];
 
-    const configuration = new Configuration({
-      basePath: randomNode.nodeApiHost,
-    });
+    let configuration: Configuration;
+    if (ctorArgs.authToken) {
```

Comment:
@elenaizaguirre It would be more flexible and less complicated to pass in the ctor args to the `Configuration` constructor as-is after having done an object merge on it with the defaults. That way you don't need the if condition at all and the caller will have total flexibility over what to pass in to the `Configuration` class' constructor.

The caller will have slightly more complicated code because they'll have to assemble the config object with the authz headers themselves, but that's a good trade-off for flexibility IMO.